tests/llm/utils/mock_toolset.py

```
@@ -367,3 +368,40 @@ def _update(self):
+    """
+    Sanitizes a URL to create a valid filename.
+    http(s)://... -> scheme is removed.
+    Characters not suitable for filenames are replaced with underscores.
+    """
+
+    # Remove scheme (http, https) if present
+    filename = re.sub(r"^https?://", "", original_file_name, flags=re.IGNORECASE)
+
+    # URL decode percent-encoded characters
+    # (e.g., %20 becomes space, %2F becomes /)
+    filename = urllib.parse.unquote(filename)
```

Flake8 (7.2.0)

[error] undefined name 'urllib'

(F821)